File: app/src/main/java/lk/chamu/stitch/ui/fragments/SignupFragment.java
package lk.chamu.stitch.ui.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import lk.chamu.stitch.R;

public class SignupFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_signup, container, false);

        // Signup Button
        View btnSignup = view.findViewById(R.id.btn_signup);
        if (btnSignup != null) {
            btnSignup.setOnClickListener(v -> 
                Navigation.findNavController(v).navigate(R.id.action_loginFragment_to_homeFragment)
            );
        }

        // Back to Login link
        TextView tvLogin = view.findViewById(R.id.tv_login);
        if (tvLogin != null) {
            tvLogin.setOnClickListener(v -> 
                Navigation.findNavController(v).navigateUp()
            );
        }

        return view;
    }
}

File: app/src/main/java/lk/chamu/stitch/ui/fragments/SignupFragment.kt
package lk.chamu.stitch.ui.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import lk.chamu.stitch.R
import android.widget.TextView

class SignupFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_signup, container, false)
        
        // Navigate back to Login
        view.findViewById<TextView>(R.id.tv_login)?.setOnClickListener {
            findNavController().navigateUp()
        }

        // Navigate to Home on signup
        view.findViewById<android.widget.Button>(R.id.btn_signup)?.setOnClickListener {
            findNavController().navigate(R.id.action_loginFragment_to_homeFragment) // Note: This action might need to be defined in nav_graph for signupFragment as well if we want direct navigation.
        }
        
        return view
    }
}
